c2corg_api/views/user_profile.py

```diff
@@ -22,9 +24,32 @@ def collection_get(self):
         return self._collection_get(
             USERPROFILE_TYPE, user_profile_documents_config)
 
-    @restricted_view(validators=[validate_id, validate_lang_param])
+    @view(validators=[validate_id, validate_lang_param])
     def get(self):
-        return self._get(UserProfile, schema_user_profile)
+        # load the requested user
+        requested_user_id = self.request.validated['id']
+        requested_user = DBSession.query(User). \
+            filter(User.id == requested_user_id). \
+            filter(User.email_validated). \
+            options(load_only(
+                User.id, User.is_profile_public, User.name, User.username)). \
+            first()
+
+        if not requested_user:
+            raise HTTPNotFound('user not found')
+        elif requested_user.is_profile_public or \
+                self.request.has_permission('authenticated'):
+            # only show the full profile if authenticated or if the user marked
+            # the profile as public
+            return self._get(UserProfile, schema_user_profile)
+        else:
+            # otherwise only return the user name
+            return {
+                'not_authenticated': True,
```

Comment:
I think ``not_authorized`` is clearer: it's the main reason why you don't see the full profile. ``not authenticated`` is the partial reason why you're not authorized (the other part being that the profile is available to authenticated users only). By the way, in the future users might choose to make their profile not available to anyone => then every request would be ``not authorized`` (even for authenticated people).